### PR TITLE
Adds Unsubscribe method on host. Listens to quit channel in subscription.

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

- Provide a description and a link to the underlying ticket

### What changes were made as part of this PR:

- Provide a high level list of the changes made

### What are the key areas to look at

- Describe the key areas for a reviewer to look at 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/host/interfaces.go [hunk 1]
```diff
@@ -22,8 +23,10 @@ type Host interface {
 	ReceiveRollup(r common.EncodedRollup)
 	// ReceiveTx processes a transaction received from a peer host.
 	ReceiveTx(tx common.EncryptedTx)
-	// CreateSubscription sets up a subscription between the host and the enclave.
-	CreateSubscription(encryptedLogSubscription common.EncryptedLogSubscription) error
+	// Subscribe sets up a log subscription between the host and the enclave.
+	Subscribe(id uuid.UUID, encryptedLogSubscription common.EncryptedLogSubscription) error
```
Comment (joeldudley) [inline go/host/interfaces.go:27]: Subscribe seemed like a better name than CreateSubscription.

File: go/host/events/backend.go [hunk 2]
```diff
@@ -72,10 +72,14 @@ func (b Backend) SubscribeRemovedLogsEvent(chan<- core.RemovedLogsEvent) event.S
 }
 
 func (b Backend) SubscribeLogsEvent(ch chan<- []*types.Log) event.Subscription {
-	logsProducer := func(<-chan struct{}) error {
+	logsProducer := func(quit <-chan struct{}) error {
 		for {
-			logs := <-b.logsCh
-			ch <- logs
+			select {
+			case <-quit:
```
Comment (joeldudley) [inline go/host/events/backend.go:78]: I need to check for the unsubscribe to not spin forever.

File: go/host/node/node.go [hunk 3]
```diff
@@ -315,18 +315,19 @@ func (a *Node) ReceiveTx(tx common.EncryptedTx) {
 	a.txP2PCh <- tx
 }
 
-// CreateSubscription sets up a subscription between the host and the enclave.
-func (a *Node) CreateSubscription(encryptedLogSubscription common.EncryptedLogSubscription) error {
```
Comment (joeldudley) [inline go/host/node/node.go:319]: I had to move the generation of the UUID to the API side, so that I know which sub ID to use for unsubscribe calls.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM - hadn't thought about filtering in WE/client being a possibility. In theory you could have two levels of filtering I guess, in WE and enclave with benefits to both.

**Pull request merged**